### DISCUSSION TITLE
How to manually attach Screenshots to a Serenity Report?

### DISCUSSION THREAD

**Discussion by afk-duck** [serenity-bdd/serenity-core#2741] [Q&A]
Hello everyone,

I am currently building a UI testautomation project for a web application.
My toolstack currently is the following:
- Cucumber as BDD-framework and for reporting
- Silk4J (version 21) for WebDriver handling and object interaction, etc.
- Junit as test runner

Currently I am taking screenshots with a Silk4J method (captures a bitmap) and successfully attaching them to the Cucumber report.
As in other non-UI projects we are using Serenity report, we also want to use it for this UI testautomation, if possible.
It is basically no problem to create the Serenity report with correct passed/failed flags.
My problem is the screenshots.
I've worked with Serenity for some time now in other projects and it is clear to me, that Serenity can only take screenshots if it handles the WebDriver.
As for this specific project I need to use Silk4J instead of Serenity for the web testing, I cannot use this simple mechanic and are forced to search for a workaround here.

So here are my questions: Is there a way to
(1) manually attach screenshots to the Serenity report?
(2) attach the screenshots to specific steps and not only to a scenario in general, as we are taking screenshots on every step?

I appreciate any help! Please let me know, if you need more info.

Best regards,
Jacky

**Comment by wakaleo** [comment]
I have no knowledge of Silk4J, but generally if you want to use Serenity for UI testing, you will need to use Serenity for your tests. Why do you need Silk4J for WebDriver?

**Reply by afk-duck** [reply]
Thanks for the fast reply. Unfortunately I am obliged to use Silk due to company policy.
I know that Serenity can do much more than just reporting, but in this case I only have the choice between staying with Cucumber report and trying to use Serenity report (I would prefer the latter).

**Comment by afk-duck** [comment]
@wakaleo Do I understand it right, that there is no provided way to manually attach a screenshot to the Serenity report?
Technically it should work, as Serenity attaches Screenshots based on the screenshot settings, right? So I guess there is some method like Serenity.attachScreenshot() (or whatever name).
It would be great if you could point me in the right direction (class name or something like that) so that I know where I can look for it.

**Reply by wakaleo** [reply]
There is no such method currently. It might be possible to add one in a future release.
